source/constructs/api/discovery_job/service.py

```diff
@@ -205,14 +216,41 @@ def start_sample_job(job_id: int, table_name: str):
         __start_sample_run(job_id, run_id, table_name)
 
 
-def __start_run(job_id: int, run_id: int):
-    job = crud.get_job(job_id)
-    run = crud.get_run(run_id)
-    run_databases = run.databases
-    if not run_databases:
-        crud.complete_run(run_id)
-        raise BizException(MessageEnum.DISCOVERY_JOB_DATABASE_IS_EMPTY.get_code(),
-                           MessageEnum.DISCOVERY_JOB_DATABASE_IS_EMPTY.get_msg())
+def __get_job_number(database_type: str) -> int:
+    if database_type in [DatabaseType.S3.value, DatabaseType.GLUE.value]:
+        return int(config_service.get_config(const.CONFIG_SUB_JOB_NUMBER_S3, const.CONFIG_SUB_JOB_NUMBER_S3_DEFAULT_VALUE))
+    return int(config_service.get_config(const.CONFIG_SUB_JOB_NUMBER_RDS, const.CONFIG_SUB_JOB_NUMBER_RDS_DEFAULT_VALUE))
+
+
+def get_run_database_ip_count(database_type: str) -> int:
+    crawler_ip = 0
+    if database_type.startswith(DatabaseType.JDBC.value):
+        crawler_ip = 3
+    return crawler_ip + __get_job_number(database_type) * 2  # Each GlueJob requires 2 IPs
+
+
+def __count_run_database_by_subnet() -> dict:
+    count_run_database = {}
+    run_databases = crud.get_running_run_databases()
+    for run_database in run_databases:
+        if not need_change_account_id(run_database.database_type):
+            continue
+        provider_id = convert_database_type_2_provider(run_database.database_type)
+        _, subnet_id = jdbc_schema.get_schema_by_real_time(provider_id, run_database.account_id, run_database.region, run_database.database_name)
+        count = count_run_database.get(subnet_id, 0)
+        count_run_database[subnet_id] = count + 1
+    logger.info(f"count_run_database:{count_run_database}")
+    return count_run_database
+
+
+def __enable_event_bridge(rule_name: str):
+    client_events = boto3.client('events')
+    client_events.enable_rule(Name=rule_name)
+
+
+def __start_run_databases(run_databases):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 88 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.